rasa/core/policies/ensemble.py

```diff
@@ -91,590 +218,316 @@ def _check_for_important_policies(self) -> None:
+    def assert_compatibility_with_domain(
+        ensemble: List[Policy], domain: Optional[Domain]
     ) -> None:
-        """Check for elements that only work with certain policy/domain combinations."""
-        policies_needing_validation = [
-            RulePolicy,
-        ]
-        for policy in policies_needing_validation:
-            policy.validate_against_domain(ensemble, domain)
-
-        _check_policy_for_forms_available(domain, ensemble)
-
-    def _check_priorities(self) -> None:
-        """Checks for duplicate policy priorities within PolicyEnsemble."""
-
-        priority_dict = defaultdict(list)
-        for p in self.policies:
-            priority_dict[p.priority].append(type(p).__name__)
-
-        for k, v in priority_dict.items():
-            if len(v) > 1:
-                rasa.shared.utils.io.raise_warning(
-                    f"Found policies {v} with same priority {k} "
-                    f"in PolicyEnsemble. When personalizing "
-                    f"priorities, be sure to give all policies "
-                    f"different priorities.",
-                    docs=DOCS_URL_POLICIES,
-                )
-
-    def _policy_ensemble_contains_policy_with_rules_support(self) -> bool:
-        """Determine whether the policy ensemble contains at least one policy
-        supporting rule-based data.
+        """Check for elements that only work with certain policy/domain combinations.
 
-        Returns:
-            Whether or not the policy ensemble contains at least one policy that
-            supports rule-based data.
-        """
-        return any(
-            policy.supported_data()
-            in [SupportedData.RULE_DATA, SupportedData.ML_AND_RULE_DATA]
-            for policy in self.policies
-        )
-
-    @staticmethod
-    def _training_trackers_contain_rule_trackers(
-        training_trackers: List[DialogueStateTracker],
-    ) -> bool:
-        """Determine whether there are rule-based training trackers.
+        # TODO: replace the ensemble (List[Policy]) with a list of policy classes and
+        # their configurations
+        # TODO: add tests once all policies have been migrated
+        # TODO: make sure this check is also called if there is just a single policy
+        # and *no* ensemble node in the graph
 
         Args:
-            training_trackers: Trackers to inspect.
+            ensemble: list of policies
+            domain: a domain
 
-        Returns:
-            Whether or not any of the supplied training trackers contain rule-based
-            data.
+        Raises:
+            `InvalidDomain` exception if the given domain is incompatible with the
+            given ensemble
         """
-        return any(tracker.is_rule_tracker for tracker in training_trackers)
+        # TODO: does it make sense to add `validate_against_domain`
+        # to other policies?
+        RulePolicy.validate_against_domain(ensemble, domain)
```

Comment:
I think this should be done by the `RulePolicy` itself during training

Reply:
nice, yeah that makes sense :)